src/array/typed_array.ts

```diff
@@ -0,0 +1,93 @@
+import { type Options, SizedType } from "../types/mod.ts";
+
+type TypedArrays =
+  | Uint8Array
+  | Uint8ClampedArray
+  | Int8Array
+  | Uint16Array
+  | Int16Array
+  | Uint32Array
+  | Int32Array
+  | Float32Array
+  | Float64Array
+  | BigUint64Array
+  | BigInt64Array;
+
+type TypedConstructors<T extends TypedArrays> = T extends Uint8Array
+  ? Uint8ArrayConstructor
+  : T extends Uint8ClampedArray ? Uint8ClampedArrayConstructor
+  : T extends Int8Array ? Int8ArrayConstructor
+  : T extends Uint16Array ? Uint16ArrayConstructor
+  : T extends Int16Array ? Int16ArrayConstructor
+  : T extends Uint32Array ? Uint32ArrayConstructor
+  : T extends Int32Array ? Int32ArrayConstructor
+  : T extends Float32Array ? Float32ArrayConstructor
+  : T extends Float64Array ? Float64ArrayConstructor
+  : T extends BigUint64Array ? BigUint64ArrayConstructor
+  : T extends BigInt64Array ? BigInt64ArrayConstructor
+  : never;
+
+export class TypedArray<T extends TypedArrays> extends SizedType<T> {
+  constructor(
+    readonly arrayConstructor: TypedConstructors<TypedArrays>,
+    readonly length: number,
+  ) {
+    super(
+      length * arrayConstructor.BYTES_PER_ELEMENT,
+      arrayConstructor.BYTES_PER_ELEMENT,
+    );
+  }
+
+  // @TODO: Not sure if this is the behavior we want
+  readUnaligned(dt: DataView, options: Options = { byteOffset: 0 }): T {
+    super.rangeCheck(dt.byteLength, options.byteOffset);
+
+    const value = new this.arrayConstructor(
+      dt.buffer,
+      dt.byteOffset + options.byteOffset,
+      this.length,
+    ) as T;
+
+    super.incrementOffset(options);
+    return value;
+  }
```

Comment:
This creates a view into the `DataView` AB instead of a copy which is what I think we want here. Same goes for the rest of this file.